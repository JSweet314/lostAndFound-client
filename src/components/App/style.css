.app {
  height: 100%;
  width: 100%;
  background: url('../../images/lonelyShoe.jpg') center no-repeat;
  background-size: cover;
}