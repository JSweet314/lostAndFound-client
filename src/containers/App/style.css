.app {
  height: 100vh;
  width: 100vw;
  background: url('../../images/background-lostAndFound.jpg') center no-repeat;
  background-size: cover; 
}