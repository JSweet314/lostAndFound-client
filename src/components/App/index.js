import React from 'react';
import { Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import UserFormContainer from '../../containers/UserFormContainer';
import ReportFormContainer from '../../containers/ReportFormContainer';
import Header from '../Header';
import LandingPage from '../LandingPage';
import ItemsContainer from '../../containers/ItemsContainer';
import './style.css';

const App = ({ username, loggedIn, handleLogOut }) => {
  return (
    <div className="app">
      <Header
        loggedIn={loggedIn}
        username={username}
        handleLogOut={handleLogOut} />
      <Switch>
        <Route exact path='/' render={() =>
          <LandingPage username={username} />} />
        <Route path='/user-forms/:id' component={UserFormContainer} />
        <Route path='/report/:id' component={ReportFormContainer} />
        <Route path='/items' component={ItemsContainer} />
      </Switch>
    </div>
  );
};

App.propTypes = {
  username: PropTypes.string,
  loggedIn: PropTypes.bool.isRequired,
  handleLogOut: PropTypes.func.isRequired
};

export default App;