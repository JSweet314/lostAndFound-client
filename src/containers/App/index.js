import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route, withRouter } from 'react-router-dom';
import Header from '../../components/Header';
import FormsContainer from '../FormsContainer';
import LandingPage from '../../components/LandingPage';
import PropTypes from 'prop-types';
import * as actions from '../../actions';
import './style.css';

export class App extends Component {

  componentDidMount = () => {
    const storedUser = localStorage.getItem('LFUser');
    if (storedUser) {
      const user = JSON.parse(storedUser);
      this.props.captureUser(user);
    }
  }

  handleLogOut = () => {
    this.props.logOutUser();
    localStorage.clear();
  }

  render() {
    const { loggedIn, username } = this.props;
    return (
      <div className="app">
        <Header 
          loggedIn={loggedIn} 
          username={username}
          handleLogOut={this.handleLogOut} />
        <Route exact path='/' render={() =>
          <LandingPage username={username} />} />
        <Route path="/forms/:id" component={FormsContainer} />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  loggedIn: state.user.loggedIn,
  username: state.user.username
});

const mapDispatchToProps = dispatch => ({
  logOutUser: () => dispatch(actions.logOutUser()),
  captureUser: user => dispatch(actions.captureUser(user))
});

App.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  username: PropTypes.string,
  logOutUser: PropTypes.func.isRequired,
  captureUser: PropTypes.func.isRequired
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));