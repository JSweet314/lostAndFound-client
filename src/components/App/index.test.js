import React from 'react';
import {shallow} from 'enzyme';
import App from './index';

describe('App', () => {
  const mockHandleLogOut = jest.fn();
  it('should match a snapshot', () => {
    const wrapper = shallow(
      <App 
        username='jon'
        loggedIn={true}
        handleLogOut={mockHandleLogOut}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});