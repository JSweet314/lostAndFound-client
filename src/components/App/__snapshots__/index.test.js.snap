// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    handleLogOut={[Function]}
    loggedIn={true}
    username="jon"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          handleLogOut={[Function]}
          loggedIn={true}
          username="jon"
/>,
        <Switch>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    path="/user-forms/:id"
          />
          <Route
                    component={[Function]}
                    path="/report/:id"
          />
          <Route
                    component={[Function]}
                    path="/items"
          />
</Switch>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleLogOut": [Function],
          "loggedIn": true,
          "username": "jon",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              path="/user-forms/:id"
/>,
            <Route
              component={[Function]}
              path="/report/:id"
/>,
            <Route
              component={[Function]}
              path="/items"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/user-forms/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/report/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/items",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            handleLogOut={[Function]}
            loggedIn={true}
            username="jon"
/>,
          <Switch>
            <Route
                        exact={true}
                        path="/"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        path="/user-forms/:id"
            />
            <Route
                        component={[Function]}
                        path="/report/:id"
            />
            <Route
                        component={[Function]}
                        path="/items"
            />
</Switch>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleLogOut": [Function],
            "loggedIn": true,
            "username": "jon",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                path="/user-forms/:id"
/>,
              <Route
                component={[Function]}
                path="/report/:id"
/>,
              <Route
                component={[Function]}
                path="/items"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/user-forms/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/report/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/items",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
